### ISSUE TITLE
Is it possible to programatically generate pages?

### ISSUE THREAD

**Issue by chrisdmacrae** [11ty/eleventy#126]
In short -- I'm looking to generate a couple variants of a single page, and I'd prefer not to do it as separate static HTML pages than w/ Javascript.

So there would be a single html file with an "empty" state and an "edit" state. The layout is slightly different, but the content/data stays the same.

Is this possible in any way, w/o having to source files?

**Comment by chrisdmacrae** [comment]
My use-case is fairly complicated. OTL

I'm generating a multi-step form, with an input and edit state. From the master `index.html`, I'm generating the form's flow with pagination already, using content chunks sourced from other files.

So...

```
- form/
--- index.html => using pagination here
--- parts/
----- part-one.html => I want to generate 2n versions of these for each state
----- part-two.html
----- part-three.html
```

`form/index.html`
```
---
pagination:
  data: parts
  size: 1
parts:
- part-one
- part-two
- part-three
---
// here I fetch the actual page by looping over a collection and using it's fileSlug.
```

So I'm thinking I can use pagination on each `part-x.html` to generate out the variants.

---

TL;DR: eleventy is the bomb!
